### PR TITLE
Add a test for r# identifiers

### PR DESCRIPTION
**Pull request by poliorcetics**
I'm not entirely sure I properly ran the test locally (I think so though), waiting for CI to confirm. :)

@rustbot modify labels: T-rustdoc

r? @jyn514

### CODE REVIEW

File: src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// ignore-tidy-linelength
+
+pub mod internal {
+    pub struct r#mod;
+
+    /// See [name], [other name]
+    ///
+    /// [name]: mod
+    /// [other name]: ../internal/struct.mod.html
```
Comment (jyn514) [inline src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs:9]: What is this meant to test?
```suggestion
    /// [other name]: crate::internal::mod
```

File: src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// ignore-tidy-linelength
+
+pub mod internal {
+    pub struct r#mod;
```
Comment (jyn514) [inline src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs:4]:
```suggestion
    // @has 'raw_ident_eliminate_r_hashtag/internal/struct.mod.html'
    pub struct r#mod;
```

File: src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs (hunk 1 above)
Comment (poliorcetics) [reply src/test/rustdoc/raw-ident-eliminate-r-hashtag.rs:9]: I wrote it too quickly and simply forgot to move it to intra-doc links. I wanted to test both the short and long versions.

**Comment by jyn514** [comment]
```
 11: @has check failed
	`XPATH PATTERN` did not match
	    // @has 'raw_ident_eliminate_r_hashtag/internal/struct.B.html' '//a[@href="../raw_ident_eliminate_r_hashtag/internal/struct.mod.html"]' 'name'
12: @has check failed
	`XPATH PATTERN` did not match
	    // @has 'raw_ident_eliminate_r_hashtag/internal/struct.B.html' '//a[@href="../raw_ident_eliminate_r_hashtag/internal/struct.mod.html"]' 'other name'
19: @has check failed
	File does not exist 'raw_ident_eliminate_r_hashtag/struct.A.html'
	// @has 'raw_ident_eliminate_r_hashtag/struct.A.html' '//a[@href="../raw_ident_eliminate_r_hashtag/internal/struct.mod.html"]' 'name'
```
Looks like maybe this doesn't behavior as expected?

**Comment by poliorcetics** [comment]
I think the XPATH is wrong, I will try something else based on existing tests in the other files.

**Comment by jyn514** [comment]
@bors r+ rollup

Thanks for the test!

**Pull request merged**